programs/tests/mt-c-tests/heap/Makefile

```diff
@@ -0,0 +1,5 @@
+FLEXPRET_ROOT_DIR = ../../../../
+NAME = heap
+APP_SOURCES = *.c
+TEST_ENVIRONMENT = true
```

Comment:
`TEST_ENVIRONMENT` ends up setting `FP_TEST_OVERRIDE` to `__attribute__(("weak"))`, which allows us to override `__malloc_lock()` and `__malloc_unlock()` in the next file. When `TEST_ENVIRONMENT` is false or not set, trying to override the malloc functions will yield a compile error.

Reply:
Why do we want to override `__malloc_lock` and `__malloc_unlock`?

Reply:
Just to be able to test it elsewhere. See the link. Here, if the user changes `#if 0` to `#if 1`, we override the implementations with no actual lock, and malloc stops working. So this shows it actually does something in the multi-threaded case.

https://github.com/pretis/flexpret/blob/context-switch/programs/tests/mt-c-tests/heap/heap.c

But when this test is not run, there is no reason for overriding these two functions. That's why the weak attribute is conditionally added as well